desktop/Makefile.am

```diff
@@ -25,23 +25,109 @@
 # OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
 # ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-include $(top_srcdir)/mk/layout.am
-
-dist_desktop_DATA = org.fontforge.FontForge.desktop
-dist_metainfo_DATA = org.fontforge.FontForge.appdata.xml
-dist_hicolor16x16_DATA = icons/16x16/apps/org.fontforge.FontForge.png
-dist_hicolor22x22_DATA = icons/22x22/apps/org.fontforge.FontForge.png
-dist_hicolor24x24_DATA = icons/24x24/apps/org.fontforge.FontForge.png
-dist_hicolor32x32_DATA = icons/32x32/apps/org.fontforge.FontForge.png
-dist_hicolor48x48_DATA = icons/48x48/apps/org.fontforge.FontForge.png
-dist_hicolor64x64_DATA = icons/64x64/apps/org.fontforge.FontForge.png
-dist_hicolor128x128_DATA = icons/128x128/apps/org.fontforge.FontForge.png
-dist_hicolor256x256_DATA = icons/256x256/apps/org.fontforge.FontForge.png
-dist_hicolorscalable_DATA = icons/scalable/apps/org.fontforge.FontForge.svg
-dist_mimepackages_DATA = fontforge.xml
-
-EXTRA_DIST = icons/src/icon-16x16-apps-fontforge.svg	\
-	icons/src/icon-24x24-apps-fontforge.svg		\
-	icons/src/icon-32x32-apps-fontforge.svg
+PACKAGEROOTDIR = ${datarootdir}
+
+# Where .desktop files go.
+desktopdir = $(PACKAGEROOTDIR)/applications
+desktop_DATA = org.fontforge.FontForge.desktop
+
+appdatadir = $(PACKAGEROOTDIR)/appdata
```

Comment:
What's going on here? In the old makefile, `org.fontforge.FontForge.appdata.xml` got installed to the metainfo folder, but now it gets installed to appdata instead? The AppImage packager expects the former.

Reply:
FTR: https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html mentions `appdata` being legacy and `metainfo` being the directory du jour.
This may be related to [this change](https://github.com/fontforge/fontforge/pull/3387#discussion_r244451928) I haven't understood.

Reply:
On older distro installs, everything is in appdata, should continue to keep appdata in there.
I'll clone another copy into metainfo too.
I've looked at a couple of recent distro /usr/share/metainfo and it's sort of a jumble of both types (bit of a mess in there).

Also removing the 'Extra_DIST = desktop' I added in the main Makefile.am as this isn't the right way to 'make dist' anyways.

Reply:
@muelli - directory du jour is a good description. Some specs seems to shift with time :-)
To avoid having to re-install, you may want to look at a couple of live distros inside those directories, I'd suggest a couple that have been around for quite a while.... look at knoppix 6, puppy linux, then compare it with a couple of recent ones, like mageia 7, ubuntu, etc. This way you avoid installing unnecessary programs but at the same time get a lay-of-the-land.